python/eos/analysis_file.py

```diff
@@ -425,7 +277,24 @@ def validate(self):
                     known_params[param]
                 except RuntimeError:
                     messages.append(f"Error in prediction {p_name}: Fixed parameter '{param}' not known to EOS")
-
+        for step_id, step in self._steps.items():
+            # Check that all the dependencies of a step exist
+            if (unknown_dependencies := step.depends_on - self._steps.keys()):
+                raise ValueError(f'Step \'{step_id}\' depends on unknown steps: {unknown_dependencies}')
```

Comment:
Hm. Shouldn't we append this to the message rather than raising an error?

Reply:
Yes